public/src/utils.js

```diff
@@ -487,14 +490,50 @@
 			});
 		},
 
-		// https://github.com/sindresorhus/is-absolute-url
-		isAbsoluteUrlRE: /^[a-zA-Z][a-zA-Z\d+\-.]*:/,
-		isWinPathRE: /^[a-zA-Z]:\\/,
-		isAbsoluteUrl: function (url) {
-			if (utils.isWinPathRE.test(url)) {
-				return false;
+		urlToLocation: function (url) {
+			return utils.parseUrl(url, true, true);
+		},
+
+		parseUrl: function (url, parseQueryString, slashesDenoteHost) {
```

Comment:
Recommend returning a runtime-independent object from this function instead of using isBrowser multiple places.

For instance, returning an object with booleans isSchemeAbsolute, isProtocolAbsolute etc

Reply:
well, I would have to "parse" twice each time, since when you're testing for scheme-absolute you would have to set `slashesDenoteHost=true` and would have to strip the leading slashes

```javascript
isProtocolAbsoluteUrl: function (url) {
    url = url.replace(/^\/{3,}/, '//');
    var a = utils.parseUrl(url, true, **true);
```

We don't need to do that when testing for scheme-absolute, so in order to get both results each time, we need to `require('url').parse()` twice

Which is doable I guess, but just letting you know.